plop-templates/core/Component.js.hbs

```diff
@@ -0,0 +1,28 @@
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import {
+  space,
+  themeGet
+} from 'styled-system'
+import {
+  mapProps,
+  applyVariations,
+  getPaletteColor
+} from './utils'
+
+const {{componentName}} = mapProps(({ align, ...props }) => ({
+  ...props
+}))(styled.{{htmlTag}}`
+  // Example: Use palette colors
+  color: ${getPaletteColor('base')};
+
+  // Advanced: Apply variation style along with custom styling from theme.componentStyles[component][variation]
```

Comment:
I am all for comments to communicate, but if you choose to comment on here, I think it needs to be more expressive in plain English what it's doing?

Or else, we could ✂️ for brevity. I'd vote for ✂️comment, and encourage the code to rely on naming convention to indicate what the logic is.